src/olympia/addons/tests/test_models.py

```diff
@@ -1674,18 +1674,29 @@ def test_addon_author_delete_not_synced_to_basket_if_addon_is_deleted(
     def test_block_property(self):
         addon = Addon.objects.get(id=3615)
         assert addon.block is None
-        assert not addon.is_addon_blocklisted
 
         del addon.block
         block = Block.objects.create(
             guid=addon.guid, updated_by=user_factory())
-        assert addon.block == addon.block
-        assert addon.is_addon_blocklisted
+        assert addon.block == block
 
         del addon.block
         block.update(guid='not-a-guid')
         assert addon.block is None
-        assert not addon.is_addon_blocklisted
+
+    def test_blocksubmission_property(self):
+        addon = Addon.objects.get(id=3615)
+        assert addon.blocksubmission is None
+
+        del addon.blocksubmission
+        submission = BlockSubmission.objects.create(
+            input_guids=addon.guid, updated_by=user_factory())
+        assert addon.blocksubmission == submission
+
+        del addon.blocksubmission
+        submission.update(input_guids='not-a-guid')
```

Comment:
Here we're updating the `submission` so it no longer includes the original `guid` (I assume). Should there also be a test where the `submission` itself is deleted, and therefore the `guid` is no longer part of a submission, or is that not something that can happen?

Reply:
They can't be deleted right now (#13278) though when the submission has finished (state changed to `SIGNOFF_PUBLISHED` or `SIGNOFF_REJECTED`) they don't, by default, get returned by `get_submissions_from_guid`